lmctl/cli/commands/lifecycledriver.py

```diff
@@ -29,17 +30,24 @@ def format_lifecycle_driver(output_format, lifecycle_driver):
 @click.option('--pwd', default=None, help='password used for authenticating with LM (only required if LM is secure and a username has been included in the environment config)')
 @click.option('--type', 'lifecycle_type', default='Ansible', help='Lifecycle type of the driver to add')
 @click.option('--url', help='url of lifecycle driver to add')
+@click.option('--certificate', help='filename of a file containing the public certificate of the lifecycle driver')
 @click.option('-f', '--format', 'output_format', default='table', help='format of output [table, yaml, json]')
-def add(environment, config, pwd, lifecycle_type, url, output_format):
+def add(environment, config, pwd, lifecycle_type, url, certificate, output_format):
     """Add a lifecycle driver"""
-    lifecycle_mgmt_driver = get_lifecycle_driver_mgmt_driver(environment, config, pwd)
-    new_lifecycle_driver = {
-        'type': lifecycle_type,
-        'baseUri': url
-    }
-    with clirunners.lm_driver_safety():
-        lifecycle_driver = lifecycle_mgmt_driver.add_lifecycle_driver(new_lifecycle_driver)
-    click.echo(format_lifecycle_driver(output_format, lifecycle_driver))
+    try:
+        lifecycle_mgmt_driver = get_lifecycle_driver_mgmt_driver(environment, config, pwd)
+        new_lifecycle_driver = {
+            'type': lifecycle_type,
+            'baseUri': url
+        }
+        if certificate is not None:
+            new_lifecycle_driver['certificate'] = read_certificate_file(certificate)
```

Comment:
Could we just wrap this line in the "try/except"?

Reply:
ok